js/ui/grid_core/ui.grid_core.column_headers.js

```diff
@@ -437,7 +407,7 @@ module.exports = {
                         rowIndex = $row[0].rowIndex,
                         column = this.getColumns(rowIndex)[cellIndex];
 
-                    return column ? this._columnsController.getVisibleIndex(column.index) : -1;
+                    return column ? column.index : -1;
```

Comment:
This change is not correct. Try to enable reordering.

Reply:
Works as expected because `getCellIndex` uses to get column index for showing header filter menu and `headerFilterController.showHeaderFilterMenu(columnIndex, false)` uses `getVisibleIndex` inside